.github/workflows/pull-request.yml

```
@@ -296,6 +273,98 @@ jobs:
+      - name: Check Apollo Server status for development
+        run: |
+          if curl -f 'http://localhost:4000/graphql?query=%7B__typename%7D' \
+          -H 'Apollo-Require-Preflight: true' | jq -e '. == {"data":{"__typename":"Query"}}' > /dev/null; then
+            echo "Apollo Server is up"
+          else
+            echo "Server is down"
+            exit 1 
+          fi
+      
+      - name: Stop the Development Server
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:2:6: Double quote to prevent globbing and word splitting

(shellcheck)